test/controllers/thinkers_controller_test.rb

```diff
@@ -1,14 +1,49 @@
 require 'test_helper'
 
 class ThinkersControllerTest < ActionController::TestCase
+  def setup
+    @thinker = Thinker.create( name: "Example Thinker", email: "thinker@example.com")
+  end
+
   test "should get new" do
     get :new
     assert_response :success
   end
 
-  #test "should get destroy" do
-   # get :destroy
-   # assert_response :success
+  test "should get index" do
+    get :index
+    assert_response :success
+    assert_not_nil assigns(:thinkers)
+  end
+
+  test "should create thinker" do
+	  assert_difference('Thinker.count') do
+	  post :create, thinker: {name: 'Some name', email: 'email@example.com'}
+	  end
+ 
+  	assert_redirected_to thinker_path(assigns(:thinker))
+	end
+
+  test "should get show" do
+  	get :show, id: @thinker.id
+  	assert_response :success
+  end
+
+  test "should get edit" do
+  	get :edit, id: @thinker.id
+  	assert_response :success
+  end
+
+  #test "should update thinker" do
+  	#put :update, id: @thinker.id
+  	#assert_response :success
   #end
 
+  test "should delete thinker" do
+  	#assert_difference('Thinker.count') do
```

Comment:
The current error in full:
```FAIL["test_should_delete_thinker", ThinkersControllerTest, 0.08780226999806473]
 test_should_delete_thinker#ThinkersControllerTest (0.09s)
        "Thinker.count" didn't change by 1.
        Expected: 4
          Actual: 2
        test/controllers/thinkers_controller_test.rb:48:in `block in <class:ThinkersControllerTest>'```

We see this because by default assert_difference expects the count to increase by 1, but in fact it decreases. You can see some docs here that show the API for this method:
http://api.rubyonrails.org/classes/ActiveSupport/Testing/Assertions.html

We can `assert_difference('Thinker.count', -1) ...` instead.